app/lib/discourse_automation/scriptable.rb

```diff
@@ -0,0 +1,116 @@
+module DiscourseAutomation
+  class Scriptable
+    attr_reader :fields, :automation
+
+    def initialize(automation)
+      @automation = automation
+      @placeholders = [:site_title]
+      @version = 0
+      @fields = []
+      @triggerables = []
+      @script = Proc.new {}
+
+      eval!
+    end
+
+    def eval!
+      public_send("__scriptable_#{automation.script.underscore}")
+      self
+    end
+
+    def name
+      @automation.script
+    end
+
+    def placeholders
+      @placeholders.uniq.compact
+    end
+
+    def placeholder(placeholder)
+      @placeholders << placeholder
+    end
+
+    def version(*args)
+      if args.present?
+        @version, = args
```

Comment:
What does this do? Extract the first element in `args` and put in it `@version`?

Reply:
yes, you could do `@version = args[0]` but I actually like this version, up to you